modules/terraform/validate_struct.go

```diff
@@ -0,0 +1,62 @@
+package terraform
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/collections"
+)
+
+type ValidationOptions struct {
+	RootDir     string
+	ExcludeDirs []string
+}
+
+// readModuleAndExampleSubDirs returns a slice strings representing the filepaths for all valid Terraform modules
+// in both the "modules" directory and "examples" directories in the project root, if they exist.
+func readModuleAndExampleSubDirs(opts ValidationOptions) ([]string, error) {
+	if opts.RootDir == "" {
+		return nil, errors.New("RootDir must be defined in ValidationOptions passed to ValidateAllTerraformModules")
+	}
+	var validationCandidates []string
+	// We want to run InitAndValidate on all valid subdirectories of both the modules and examples dirs
+	modulesDir := filepath.Join(opts.RootDir, "modules")
+	examplesDir := filepath.Join(opts.RootDir, "examples")
+
+	moduleSubDirs, readModulesErr := os.ReadDir(modulesDir)
+	if readModulesErr != nil {
+		return nil, readModulesErr
+	}
+	exampleSubDirs, readExamplesErr := os.ReadDir(examplesDir)
+	if readExamplesErr != nil {
+		return nil, readExamplesErr
+	}
```

Comment:
Does this handle nested module/example folders? E.g., the service catalog and `terraform-aws-monitoring`.

As written, I think it will fail to work for those but I might be wrong.

Reply:
Recursive walking of directories added in 9519b48